### ISSUE TITLE
EV_READ keep triggering when input buffer reaches highmark, behavior is different from 2.1.8-stable, seems a bug?

### ISSUE THREAD

**Issue by tim-le** [libevent/libevent#891]
with 2.1.8-stable,  let's say, a client socket has a input buffer with highmark 10 bytes,  when 10 bytes or more data are arrived,  input buffer has now got 10 bytes, then a event of EV_READ is triggered , and assume that we are busy doing something, **did not remove any data from input buffer**,   even though，EV_READ will not be triggered again.
however with release-2.1.11-stable,   EV_READ  will be triggered again and again , cause cpu 100%. until data in input buffer is lower than highmark.

**Comment by tim-le** [comment]
[https://github.com/libevent/libevent/commit/5a455acd3bdf9a0117f909450ac7ea693ca93912#diff-12c22fbded898944263268524bf00419](url)
related commit.

**Comment by azat** [comment]
Indeed, the attached patch changes the behaviour, because it causes
stalls.

If you are not reading the data from the callback in place when you are
planning to do this? Are you have some kind of proxy that has some
relations for the bufferevent that will drain the buffer after?

Of course you can disable EV_READ for bev, but let's figure this down.

**Comment by tim-le** [comment]
yep, for avoiding high cpu problem, I am now have to disable EV_READ.
this issue came from a proxy server based on libevent,recently I updated it with 2.1.11-stable, then found this problem.
I was downloading a file on chrome browser through proxy,  once I click pause, the proxy server got high cpu, because proxy couldn't send data to client side,the input buffer of target side quickly reached highmark.
